pydm/utilities/stylesheet.py

```diff
@@ -0,0 +1,128 @@
+
+import logging
+logger = logging.getLogger(__name__)
+from . import is_pydm_app
+
+
+# Fallback global stylesheet if there is no global stylesheet provided via env variable or command line parameter
+# TODO: Expand this global stylesheet
+GLOBAL_STYLESHEET = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../default_stylesheet.qss')
+
+
+def apply_stylesheet(stylesheet_file_path, timer=None):
+    """
+    Apply a stylesheet to the current Qt Designer form (a .ui file) or to a Designer form using the PyDM Launcher.
+
+    Parameters
+    ----------
+    stylesheet_file_path : str
+        The full path to a global CSS stylesheet file
+    timer : QTimer
+        A timer to launch the set stylesheet method for Qt Designer. For the PyDM Launcher, the timer is not needed,
+        and can be set to None
+    """
+    if not stylesheet_file_path:
+        # If there is no stylesheet path provided by a command parameter, check for the env variable
+        stylesheet_file_path = os.getenv("PYDM_STYLESHEET", None)
+
+    # Load style data from the stylesheet file. Otherwise, the fallback is already in place, i.e. PyDM will be
+    # using the data from the global stylesheet
+    style_data = _get_style_data(stylesheet_file_path)
+
+    if timer:
+        # For PyDM Launcher, the timer should be None. This code is to handle Qt Designer only
+        timer.timeout.connect(partial(_set_style_data, style_data, timer))
+        timer.start()
+    else:
+        # For running PyDM Launcher
+        _set_style_data(style_data, None)
+
+    return style_data
+
+
+def _get_style_data(stylesheet_file_path):
+    """
+    Read the global stylesheet file and provide the style data as a str.
+
+    Parameters
+    ----------
+    stylesheet_file_path : str
+        The path to the global stylesheet.
+
+    Returns
+    -------
+    The style data read from the stylesheet file : str
+    """
+    style_data = None
+    if stylesheet_file_path is not None:
+        try:
+            with open(stylesheet_file_path, 'r') as stylesheet_file:
+                logger.info("Opening style file '{0}'...".format(stylesheet_file_path))
+                style_data = stylesheet_file.read()
+        except Exception as ex:
+            logger.error("Error reading the stylesheet file '{0}'. Exception: {1}".format(stylesheet_file_path,
+                                                                                          str(ex)))
+    else:
+        try:
+            with open(GLOBAL_STYLESHEET) as default_stylesheet:
+                logger.info("Opening the default stylesheet '{0}'...".format(GLOBAL_STYLESHEET))
+                style_data = default_stylesheet.read()
+        except Exception as ex:
+            logger.error("Cannot find the default stylesheet file '{0}'. Exception: {1}".format(GLOBAL_STYLESHEET,
+                                                                                                str(ex)))
+    return style_data
```

Comment:
This will blow if somehow there is a problem reading the `stylesheet_file_path` since `style_data` will be `None` and will never get the `GLOBAL_STYLESHEET`. I suggest a change of order and always load the default one and overwrite the variable if we can with the specified stylesheet. This way PyDM will behave properly even if someone makes a mistake at the stylesheet specification somehow.